fdbserver/GrvProxyServer.actor.cpp

```diff
@@ -877,6 +864,18 @@ ACTOR static Future<Void> transactionStarter(GrvProxyInterface proxy,
 			transactionQueue->pop_front();
 			requestsToStart++;
 		}
+		if (!batchQueue.empty()) {
+			grvProxyData->stats.lastBatchQueueThrottled = true;
+			grvProxyData->stats.batchThrottleStartTime = now();
+		} else {
+			grvProxyData->stats.lastBatchQueueThrottled = false;
+		}
+		if (!defaultQueue.empty()) {
+			grvProxyData->stats.lastDefaultQueueThrottled = true;
+			grvProxyData->stats.defaultThrottleStartTime = now();
+		} else {
+			grvProxyData->stats.lastDefaultQueueThrottled = false;
+		}
```

Comment:
This looks about right to me. The one change I'd recommend is to mark that batch priority is throttled in the reply based on both default and batch queues. The main reason for this is to deal with a potential lull in the activity for a particular priority -- for example if we have no batch requests for a while we might mistakenly believe batch is unthrottled even though default priority still is.